### PR TITLE
Feature/InstructionsModal Responsive Design

### PR DESCRIPTION
**Pull request by leachtucker**
# Description
This pull request relates to the responsive design changes that were made to the Instructions Modal that appears just before a child joins a squad.

Fixes #

- What work was done? Added a media query for mobile screens using the prefined LESS variable. Added an additional LESS variable for screens <=320px in width. These media queries decrease the size of accept button and instruction text.
- Why was this work done? To ensure cohesive design patterns between various screen sizes
- What feature/user story is it for? _'It's time to join your squad' modal - you reach this after clicking on the accept the mission button on the dashboard. Responsive adjustments are needed at all sizes._

## Type of change

Please delete options that are not relevant.

- ✓ New feature (non-breaking change which adds functionality)

## Change Status

- ✓ Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- ✓ Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [✓] My code follows the style guidelines of this project
- [✓] I have performed a self-review of my own code
- [✓] My code has been reviewed by at least one peer
- [✓] I have commented my code, particularly in hard-to-understand areas
- [✓] I have made corresponding changes to the documentation
- [✓] My changes generate no new warnings
- [✓] I have added tests that prove my fix is effective or that my feature works
- [✓] New and existing unit tests pass locally with my changes
- [✓] There are no merge conflicts

### CODE REVIEW

**Comment by leachtucker** [comment]
['It's time to join your squad' modal - you reach this after clicking on the accept the mission button on the dashboard. Responsive adjustments are needed at all sizes. @tucker195](https://trello.com/c/7S732Pse/17-its-time-to-join-your-squad-modal-you-reach-this-after-clicking-on-the-accept-the-mission-button-on-the-dashboard-responsive-adj)

**Comment by leachtucker** [comment]
Merged Rob's pull requests. Added one less variable for extra-small screens. Removed inline style on component and pushed styling to associated LESS file (to maintain consistent design patterns throughout)

File: src/styles/less/variables.less [hunk 1]
```diff
@@ -13,6 +13,6 @@
 @button-blue: #007aff;
 
 // Media Queries
-
+@xs: ~'(max-width:320px)';
```
Comment (Robert-Lark) [inline src/styles/less/variables.less:16]: I really wanted to add more global media breakpoints also - especially one in the 500 range. Perhaps if it is something that becomes an issue in a few places we can talk about putting one in.

File: src/styles/less/InstructionsModal.less [hunk 2]
```diff
@@ -11,14 +11,58 @@
...
+        font-size: 3rem;
+      }
+    }
   }
 }
+
+@media @tablet {
+  .instructions-modal {
+    .ant-modal-content {
+        .accept-button {
```
Comment (Robert-Lark) [inline src/styles/less/InstructionsModal.less:31]: I see you nested your css selectors - not that this is an issue at all but the rest of the css (at least the stuff I was looking at) is written un-nested. Is there a difference in performance doing it one way over the other?

Screen Shot 2021-04-21 at 3 22 05 PM

File: src/styles/less/InstructionsModal.less [hunk 3]
```diff
@@ -11,14 +11,58 @@
...
+  .instructions-modal {
+    .ant-modal-content {
+        .ant-modal-body {
+          p {
+            font-size: 1.4rem;
+          }
+        }
+    }
+  }
+}
```
Comment (Robert-Lark) [inline src/styles/less/InstructionsModal.less:68]: Did you see any reason that this modals CSS needed its own file? Perhaps we can put it into the main file with all the other CSS for the page it pops out from. We could just separate it out with some text stating what the following CSS relates to.

**Review by Robert-Lark — APPROVED** [review]
Looks great - really nice scope, not too much code to scroll through - now can you do all the CSS for my personal project for me?

File: src/styles/less/InstructionsModal.less (hunk 2 above)
Comment (leachtucker) [reply src/styles/less/InstructionsModal.less:31]: Hey Rob, that's a good point. The reason I nested the selectors is due to my understanding of that being the best practice with LESS. In addition to that, the rest of the LESS file was written with a similar pattern. If I am mistaken about that being the best practice, please let me know!..

File: src/styles/less/InstructionsModal.less (hunk 3 above)
Comment (leachtucker) [reply src/styles/less/InstructionsModal.less:68]: At the moment, it appears to be a unique modal (as in, there are no other modals within the application that follow a similar look/layout.) Considering that, I think it's a good idea to separate it into its own LESS file as has been done. 
Furthermore, the modals that we have spoken about implementing for the Forms (update settings, child name, credit card, ETC) will follow a different layout, so I think we will have to be sure to keep that LESS styling separate from this InstructionModal if we would like to follow the current design pattern of the application.

File: src/styles/less/InstructionsModal.less (hunk 2 above)
Comment (Robert-Lark) [reply src/styles/less/InstructionsModal.less:31]: Cool - Yes, personally I have always nested my CSS as I think it is more readable that way.

Comment (leachtucker) [reply src/styles/less/InstructionsModal.less:31]: Just to clarify:

The nesting with LESS is a way to apply styles only to a parent's children.
For example, the code snippet you commented on will only apply styles to `.accept-button` classes that are children of elements with the class `.ant-modal-content`

**Review by RyanMersmann — COMMENTED** [review]
Looks good! The CSS seems consistent with the rest of the application

**Review by RyanMersmann — APPROVED** [review]

**Pull request merged**